src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -271,25 +324,22 @@ private static void addImageHiddenProperyValues(Element hiddenImageNode, Propert
     }
 
     /**
-     * Builds default node structure of dialog xml in the document passed in based on dialogType.
+     * Builds default node structure of dialog xml in the document passed in based
+     * on dialogType.
      *
      * @param document The {@link Document} object
      * @param root The root node to append children nodes to
      * @return Node
      */
     private static Node updateDefaultNodeStructure(Document document, Element root) {
-        Element containerElement = document.createElement("content");
-        containerElement.setAttribute(Constants.JCR_PRIMARY_TYPE, Constants.NT_UNSTRUCTURED);
-        containerElement.setAttribute(Constants.PROPERTY_SLING_RESOURCETYPE, Constants.RESOURCE_TYPE_CONTAINER);
+        Element containerElement = createNode(document, "content", Constants.RESOURCE_TYPE_CONTAINER);
 
         Element layoutElement1 = document.createElement("layout");
         layoutElement1.setAttribute(Constants.JCR_PRIMARY_TYPE, Constants.NT_UNSTRUCTURED);
```

Comment:
Should `createNode()` be used here as well?

Reply:
Moved the lines of code to a common code and changed where it is being used. It's the same code earlier and follows the same dialog structure as OOTB